crates/primitives-traits/src/block/header.rs

```diff
@@ -49,3 +49,53 @@ impl<T> BlockHeader for T where
         + MaybeSerde
 {
 }
+
+/// Helper trait to implement [`BlockHeader`] functionality for all [`Block`](crate::Block) types.
+pub trait Header {
```

Comment:
isn't think identical to `alloy_consensus::Header`?

Reply:
yes. not possible to impl foreign trait for generic `T: Block`.

Reply:
if this really simplifies things a lot I'd prefer those as `Block` trait methods with default implementations

although doing `block.header().number()` is not that bad imo

Reply:
if you default impl these, you lose the ability to inline them afaik. would inline all getters, ref https://github.com/alloy-rs/alloy/pull/1642

Reply:
but sure, can have identical trait methods on `Block`

Reply:
the whole point though is for rollups to have to avoid implementing lots of boilerplate code.

there is a reason we avoided calling `.header()` everywhere in the original code I believe.